### PR TITLE
Igor/infra web app znk exercise states

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: demo/workouts/index.html [hunk 1]
```diff
@@ -82,6 +82,5 @@
     <script src="storagesConfig.js"></script>
     <script src="contentConfig.js"></script>
     <script src="env.service.js"></script>
-    <script src="analyticsConfig.js"></script>
```
Comment (assafshp) [inline demo/workouts/index.html]: no need for this file?

File: src/core/module.js [hunk 2]
```diff
@@ -2,39 +2,7 @@
...
-        'znk.infra.storage',
-        'znk.infra.svgIcon',
-        'znk.infra.user',
-        'znk.infra.utility',
-        'znk.infra.znkAudioPlayer',
-        'znk.infra.znkExercise',
-        'znk.infra.znkModule',
-        'znk.infra.znkTimeline'
+        //all modules will be injected here
+        /** allModules **/
```
Comment (assafshp) [inline src/core/module.js]: Good

File: test/mock/analytics/analytics.mock.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+(function (angular) {
```
Comment (assafshp) [inline test/mock/analytics/analytics.mock.js]: why do we need this ?
why only now?

Comment (igormas) [inline test/mock/analytics/analytics.mock.js]: it protecting angular object from being overwritten

File: demo/workouts/index.html (hunk 1 above)
Comment (igormas) [inline demo/workouts/index.html]: Nope

File: src/components/znkExercise/controllers/baseZnkExercise.controller.js [hunk 4]
```diff
@@ -37,6 +34,28 @@
...
+            function _finishExercise() {
+                exerciseResult.isComplete = true;
+                exerciseResult.endedTime = Date.now();
+                exerciseResult.$save();
+
+                var exerciseTypeValue = ExerciseTypeEnum.getValByEnum(exerciseData.exerciseTypeId);
+                exerciseTypeValue = exerciseTypeValue.toLowerCase();
+                var broadcastEventName = exerciseEventsConst[exerciseTypeValue].FINISH;
+
+                $rootScope.$broadcast(broadcastEventName, exercise, exerciseResult, exerciseData.examData);
```
Comment (assafshp) [inline src/components/znkExercise/controllers/baseZnkExercise.controller.js]: don't we need to move it only after the stats is completed (next line)?

File: src/components/znkExercise/controllers/baseZnkExercise.controller.js [hunk 5]
```diff
@@ -37,6 +34,28 @@
                 return exercise.time;
             }
 
+            function _finishExercise() {
+                exerciseResult.isComplete = true;
+                exerciseResult.endedTime = Date.now();
+                exerciseResult.$save();
+
+                var exerciseTypeValue = ExerciseTypeEnum.getValByEnum(exerciseData.exerciseTypeId);
```
Comment (assafshp) [inline src/components/znkExercise/controllers/baseZnkExercise.controller.js]: why not ExerciseTypeEnum.getValByEnum(exerciseData.exerciseTypeId).tolowercase()?

Comment (igormas) [inline src/components/znkExercise/controllers/baseZnkExercise.controller.js]: too long

File: src/components/znkExercise/controllers/baseZnkExercise.controller.js (hunk 4 above)
Comment (igormas) [inline src/components/znkExercise/controllers/baseZnkExercise.controller.js]: Fixed

**Pull request merged**